# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/abort/AbortSignal.cpp
# Commit: 289709006a92
# Full Hash: 289709006a928c950041021bec8e1567d41d473e
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2024-07-22 21:46:43
# Description:
#   Bug 1908466 - Ensure source signal is not null when creating a dependent signal. r=saschanaz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D217233
# ==============================================================================

diff -r 95624d3e6b46 -r 289709006a92 dom/abort/AbortSignal.cpp
--- a/dom/abort/AbortSignal.cpp	Mon Jul 22 11:00:38 2024 +0000
+++ b/dom/abort/AbortSignal.cpp	Mon Jul 22 12:32:06 2024 +0000
@@ -293,6 +293,12 @@
     } else {
       // Step 4.2. Otherwise, make resultSignal dependent on its source signals
       for (const auto& sourceSignal : signal->mSourceSignals) {
+        if (!sourceSignal) {
+          // Bug 1908466, sourceSignal might have been garbage collected.
+          // As signal is not aborted, sourceSignal also wasn't.
+          // Thus do not depend on it, as it cannot be aborted anymore.
+          continue;
+        }
         MOZ_ASSERT(!sourceSignal->Aborted() && !sourceSignal->Dependent());
         resultSignal->MakeDependentOn(sourceSignal);
       }